app/controllers/projects_controller.rb

```diff
@@ -22,6 +22,14 @@ class ProjectsController < ApplicationController
   EDIT_ACTIONS = [:edit, :update, :destroy, :add_user, :all_users, :update_project_visibility].freeze
   OTHER_ACTIONS = [:create, :new, :index, :send_project_csv, :choose_project].freeze
 
+  # Required for token auth for CLI actions
+  skip_before_action :verify_authenticity_token, only: [:index, :create]
+  before_action :authenticate_user!, except: [:index, :create]
+  acts_as_token_authentication_handler_for User, only: [:index, :create], fallback: :devise
+  current_power do
```

Comment:
Nit: this initialization of current_power is already in `ApplicationController`. Do we need it here again?

Reply:
Weirdly enough it didn't work until I added this (and same was in samples_controller) as

```
  current_power do # Put this here for CLI
    Power.new(current_user)
  end
```

I guess it doesn't go through the same steps for API calls